File: Java/T7/codigos2021/T07_ViewPager/app/src/main/java/com/example/t07_viewpager/FragmentDos.java
package com.example.t07_viewpager;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class FragmentDos extends Fragment {

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        Log.v("fragment","OnAttach 2");
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Log.v("fragment","OnDettach 2");


    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.layout_fragment_dos, container,false);
        return view;
    }
}

File: Kotlin/T5/codigos_ces/T6_Fragments/app/src/main/java/com/example/t6_fragments/ui/fragments/FragmentDos.kt
package com.example.t6_fragments.ui.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.t6_fragments.databinding.FragmentDosBinding

class FragmentDos: Fragment() {

    private lateinit var context: Context
    private lateinit var binding: FragmentDosBinding
    private lateinit var nombreRecuperado: String;

    companion object{
        fun newInstance(nombre: String): FragmentDos {
            val fragmentDos: FragmentDos = FragmentDos()
            val bundle: Bundle = Bundle()
            bundle.putString("nombre",nombre)
            fragmentDos.arguments = bundle
            return fragmentDos;
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.context = context;
         nombreRecuperado = arguments?.getString("nombre") ?: "sin nombre"
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentDosBinding.inflate(layoutInflater,container,false)
        return binding.root
    }

    override fun onStart() {
        super.onStart()
        binding.textoNombre.text = nombreRecuperado
    }

    override fun onDetach() {
        super.onDetach()
    }
}